### PR TITLE
feat: Include HealthCheck in server pkg

### PR DESCRIPTION
**Pull request by wsalles**
Closes [PLT-56]

[PLT-56]: https://ydata.atlassian.net/browse/PLT-56?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pkg/common/server/server.go [hunk 1]
```diff
@@ -85,3 +85,8 @@ func (s *Server) RunSecurely(ctx context.Context) {
 		s.log.Infof("Server exiting")
 	}()
 }
+
+// UseHealthCheck creates a new HealthCheck route
+func (s *Server) UseHealthCheck() {
```
Comment (renatoaguimaraes) [inline pkg/common/server/server.go:90]: The health check is optional? I think we can call at Server constructor.
For k8s applications the probes are very important, both /healthz for Liveness and /readyz for Readiness.
For example, for readiness we can create a method with a func parameter thats need to implemented for each service.

**Review by renatoaguimaraes — APPROVED** [review]

**Review by renatoaguimaraes — CHANGES_REQUESTED** [review]
Let's think about readyz and healthz

File: pkg/common/server/middlewares.go [hunk 2]
```diff
@@ -44,3 +44,26 @@ func (s Server) setUserID() gin.HandlerFunc {
...
+			return
+		}
+		ctx.Status(http.StatusNoContent)
+	}
+}
+
+// SetReadyzState enables/disables readyz
+func (s *Server) SetReadyzState(state bool) {
+	s.readyzAvailable.Store(state)
+}
```
Comment (portellaa) [inline pkg/common/server/middlewares.go:69]: well, perhaps they don't fit very well as middleware's

**Review by portellaa — COMMENTED** [review]
We could do something as `AddHealthz` and `AddReadyz` this could be enabled by the user if he wants.
I can see we needing this all the time, but i think it doesn't hurt to have this configurable

File: pkg/common/server/server.go [hunk 3]
```diff
@@ -48,40 +50,58 @@ func NewServer(log logging.Logger, configuration HTTPServerConfiguration) *Serve
...
+			s.logger.Fatalf("Server forced to shutdown: %v", err)
 		}
 
-		s.log.Infof("Server exiting")
+		s.logger.Infof("Server exiting")
 	}()
 }
+
+// AddHealthz creates a route to LivenessProbe
+func (s *Server) AddHealthz() {
```
Comment (portellaa) [inline pkg/common/server/server.go:92]: are we receiving an endpoint here if want to override?

File: pkg/common/server/server.go [hunk 4]
```diff
@@ -48,40 +50,58 @@ func NewServer(log logging.Logger, configuration HTTPServerConfiguration) *Serve
...
+	s.Router.GET(s.configuration.HealthzEndpoint, s.healthz())
+}
+
+// AddReadyz creates a route to ReadinessProbe
+func (s *Server) AddReadyz() {
```
Comment (portellaa) [inline pkg/common/server/server.go:97]: same as healthz and the callback

**Review by portellaa — COMMENTED** [review]
mixed feelings about the names for AddHealthz and AddReadyz 🤔 
perhaps something different, what do you think @renatoaguimaraes @wsalles ?

File: pkg/common/server/server.go (hunk 3 above)
Comment (wsalles) [reply pkg/common/server/server.go]: about the name, I liked it!
now the endpoint is highly configurable. please, see the latest changes and tell me what do you think?

File: pkg/common/server/server.go (hunk 4 above)
Comment (wsalles) [reply pkg/common/server/server.go]: callback also to healthz? 🤔 
is it necessary?

File: pkg/common/helpers/utils.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	for _, value := range list {
+		if key == value {
+			return true
+		}
+	}
+	return false
+}
+
+// FirstStringOfArrayWithFallback is a handle to get the first position of the array with fallback
+func FirstStringOfArrayWithFallback(list []string, fallback ...string) string {
```
Comment (portellaa) [inline pkg/common/helpers/utils.go:14]: why is the fallback a cvarg?

File: pkg/common/helpers/utils.go [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+// ArrayContainsString is a helper to find out if a given string is inside an array or not
+func ArrayContainsString(list []string, key string) bool {
+	for _, value := range list {
+		if key == value {
+			return true
+		}
+	}
+	return false
+}
```
Comment (portellaa) [inline pkg/common/helpers/utils.go:11]: why do we need this?

File: pkg/common/helpers/utils.go [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+// LastStringOfArrayWithFallback is a handle to get the last position of the array with fallback
+func LastStringOfArrayWithFallback(list []string, fallback ...string) string {
+	if len(list) > 0 {
+		return list[len(list)-1]
+	} else if len(fallback) > 0 {
+		return fallback[0]
+	}
+	return ""
+}
```
Comment (portellaa) [inline pkg/common/helpers/utils.go:31]: and this?

File: pkg/common/helpers/utils.go [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package helpers
+
+// FirstStringOfArrayWithFallback is a handle to get the first position of the array with fallback
+func FirstStringOfArrayWithFallback(list []string, fallback string) string {
```
Comment (portellaa) [inline pkg/common/helpers/utils.go:4]: i would move this method to the server and make it private, don't see we needing it

Comment (wsalles) [reply pkg/common/helpers/utils.go]: done.

**Review by portellaa — APPROVED** [review]

**Review by renatoaguimaraes — APPROVED** [review]

**Pull request merged**